cmd/minikube/cmd/config/config_test.go

```diff
@@ -41,12 +41,14 @@ var configTestCases = []configTestCase{
     "ReminderWaitPeriodInHours": 99,
     "cpus": 4,
     "disk-size": "20g",
+    "iso-url": "http://foo.bar/minishift-centos.iso",
     "log_dir": "/etc/hosts",
     "show-libmachine-logs": true,
     "v": 5,
     "vm-driver": "kvm"
 }`,
 		config: map[string]interface{}{
+			"iso-url": "http://foo.bar/minishift-centos.iso",
```

Comment:
> The set/get functionality seems to just deal with MinikubeConfig. Can there be other data? If not, does this code throw an error?

What do you mean other data, can you provide some pointer?

> What if there is a mis-typed flag? Can we extend these tests a bit to cover some border cases? 

Set method already have test which tell if we put a unknown property and it does throw an error.

```
$ ./minikube config set foo bar
Cannot find property name foo
```

> Also we need to better understand what happens, if the user manually edits these files and maybe introduces typos.

If a user manually edit then minishift doesn't care about extra flags or if by mistake user introduced a typo in config that also ignored by minishift.

```
# Manually edited config file and introduced a typo and also another variable which is not
# part of property
$ ./minikube config view
- memory: 4096
- vm-drive: virtualbox
- foo: bar
- iso-url: file:/Users/prkumar/minishift-centos.iso

# minishift started ignoring typo and extra variable and used default values.
$ ./minishift start --show-libmachine-logs
Starting local OpenShift cluster using 'xhyve' hypervisor...
Running pre-create checks...
Creating machine...
...
```